lib/src/utils/txDecoder.ts

```
@@ -73,77 +79,276 @@ export class TxDecoder {
+        const txObject = JSON.parse(txString);
+        const decodedTx: Tx = assertAndReturnValidTx(txObject);
 
-        const parsedTxBody = JSON.parse(txBodyStringified);
-        const obj = { ...parsedTxBody };
-        obj.messages = txBody.messages.map(({ typeUrl, value }) => {
-            if (!typeUrl) {
-                throw new Error('Missing type_url in Any');
-            }
-            if (!value) {
-                throw new Error('Missing value in Any');
+        if (!decodedTx.authInfo || !decodedTx.body) {
+            throw new Error('Provided JSON is invalid.');
+        }
+
+        // Todo: Looks like we need to support `nonCriticalExtensionOptions` and `extensionOptions`
+        if (decodedTx.body.nonCriticalExtensionOptions.length > 0 || decodedTx.body.extensionOptions.length > 0) {
+            throw new Error("JSON Decoder doesn't support 'nonCriticalExtensionOptions' or 'extensionOptions'");
+        }
+
+        /**
+         * Creating a RawTransaction instance
+         *
+         */
+        const croSdk = CroSDK({ network });
+        const rawTx = new croSdk.RawTransaction();
+
+        /**
+         * Lower section handles `authInfo` related values derivation
+         *
+         */
+        const gasLimitString = decodedTx.authInfo.fee?.gasLimit.toString(10)!;
+
+        // Default `Fee` used as ZERO(0)
+        let feeCoin = croSdk.Coin.fromBaseUnit('0');
+
+        if (
+            (decodedTx.authInfo.fee?.amount && decodedTx.authInfo.fee?.amount.length > 1) ||
+            decodedTx.authInfo.fee?.amount.length! < 1
+        ) {
+            // @todo: revisit this in IBC
+            throw new Error('Invalid fee amount provided.');
+        }
+
+        //  Note: Considering only first element as we support non-array mode
+        const feeAmountString = decodedTx.authInfo.fee?.amount[0]!.amount!;
+        const feeAmountDenom = decodedTx.authInfo.fee?.amount[0]!.denom;
```

Have we handled it in next PR. To avoid users unaware of the losing fee. Should we throw an Error if the array length > 1?

---

I have included a throw check a few lines above this code block.